homepage/tests.py

```diff
@@ -88,3 +89,42 @@ def test_multiple_para_rendering_and_NO_unsafe_tags(self):
     def test_TOC(self):
         response = self.client.post(reverse('homepage:create'), self.input1, follow=True)
         self.assertContains(response, '<ul><li>will</li></ul><ul><ul><li>hi</li></ul></ul>', status_code=200)
+
+
+class LogInTest(TestCase):
+    def setUp(self):
+        self.input = {
+            'username': 'testuser',
+            'password': 'secret'}
+        self.reg = {
+            'username': 'testuser1',
+            'password1': 'secret.811',
+            'password2': 'secret.811'}
+        self.input2 = {
+            'username': 'testuser3',
+            'password': 'secret'}
+        User.objects.create_user(**self.input)
+
+    def test_login(self):
+        response = self.client.post(reverse('homepage:login'), self.input, follow=True)
```

Comment:
Django's TestClient has a login method which returns True if the user was successfully logged in.
https://docs.djangoproject.com/en/stable/topics/testing/tools/#django.test.Client.login